spec/requests/users/invitation_spec.rb

```diff
@@ -1,153 +1,186 @@
           end
 
           it 'creates a new user for each supplied email address' do
-            newest_users = User.order(invitation_sent_at: :desc).last(2)
-            emails = []
-            newest_users.each { |u| emails << u.email }
-            expect(emails.sort).to eq(friends_array.sort)
+            # Arrange
+            emails = User.order(invitation_sent_at: :desc).last(friends_array.count).pluck(:email)
+
+            # Assert
+            expect(emails).to match_array(friends_array)
           end
 
           it 'sends an invitation email to each valid address' do
+            # Arrange
             emails = []
+
+            # Act
             Devise.mailer.deliveries.each { |d| emails << d.to[0] }
-            expect(Devise.mailer.deliveries.count).to eq(2)
-            expect(emails.sort).to eq(friends_array.sort)
+
+            # Assert
+            expect(Devise.mailer.deliveries.count).to eq(friends_array.count)
+            expect(emails).to match_array(friends_array)
           end
 
           it 'creates a unique token for each email' do
+            # Arrange
             friend1 = User.find_by(email: invitee1)
             friend2 = User.find_by(email: invitee2)
+
+            # Assert
             expect(friend1.invitation_token).not_to eq(friend2.invitation_token)
           end
         end
       end
 
       context 'when invalid params are given' do
-        let(:invalid_invite) { post :create, params: { user: { email: invalid_email } } }
+        let(:invalid_invite) {
+          post user_invitation_path, params: { user: { email: invalid_email } }
+        }
 
         it 're-renders the invitation form' do
+          # Act
           invalid_invite
+
+          # Assert
           expect(response).to redirect_to new_user_invitation_path
         end
 
         it 'does not create a new user for an invalid email' do
-          user_count = User.count
-          invalid_invite
-          expect(User.count).to eq(user_count)
+          expect { invalid_invite }.not_to change { User.count }
         end
       end
 
       context 'when both valid and invalid params are given' do
-        before(:each) do
-          post :create, params: { user: { email: "#{invitee1}, #{invalid_email}" } }
-        end
+        let(:valid_and_invalid_invite) {
+          post user_invitation_path, params: { user: { email: "#{invitee1}, #{invalid_email}" } }
+        }
 
         it 'only creates a new User for the valid email' do
-          newest_users = User.order(invitation_sent_at: :desc).last(2)
-          emails = []
-          newest_users.each { |u| emails << u.email }
-          expect(emails.sort).to include(invitee1)
-          expect(emails.sort).not_to include(invalid_email)
+          # Assume
+          emails = User.pluck(:email)
+          expect(emails).not_to include(invitee1)
+          expect(emails).not_to include(invalid_email)
```

Comment:
I added an **Assume** step here  to make sure that there aren't any users with `invitee1` and `invalid_email` emails before `create` invitations path was called.